src/FffGcodeWriter.cpp

```diff
@@ -2015,14 +2015,17 @@ bool FffGcodeWriter::partitionInfillBySkinAbove(Polygons& infill_below_skin, Pol
         const size_t skin_layer_nr = gcode_layer.getLayerNr() + i;
         if (skin_layer_nr < mesh.layers.size())
         {
-            for (const SliceLayerPart& part : mesh.layers[skin_layer_nr].parts)
+            for (const SliceLayerPart& part_i : mesh.layers[skin_layer_nr].parts)
             {
-                for (const SkinPart& skin_part : part.skin_parts)
+                for (const SkinPart& skin_part : part_i.skin_parts)
                 {
+                    // Limit considered areas to the ones that should have infill underneath at the current layer.
+                    const Polygons relevant_outline = skin_part.outline.intersection(part.getOwnInfillArea());
```

Comment:
Isn't `part.getOwnInfillArea()` always a strict subset of `skin_part.outline`?

Reply:
Not really, because these needn't be on the same layer. The variable `part` (and note that I renamed the one in the loop to something else, to prevent the shadowing that was going on) is the one for the 'current' layer, always, and the skin-part outlines are in the layers above (`i` is always > 0, so it doesn't even touch the current layer).

If you went with only the infill, the skin-support-edges wouldn't be underneath the skin, which would defeat the purpose.